File: Modules/Sessions/Http/Resources/SessionResource.php
<?php

namespace Modules\Sessions\Http\Resources;

use App\Settings\GeneralSettings;
use Illuminate\Http\Resources\Json\JsonResource;
use Modules\Auth\Http\Resources\UserResource;
use Modules\Groups\Http\Resources\GroupResource;
use Modules\Sessions\Enums\SessionStatus;
use Modules\Sessions\Enums\SessionType;

class SessionResource extends JsonResource
{
    public function toArray($request): array
    {
        $settings = app(GeneralSettings::class);

        $user = $request->user();
        $isBooked = $user ? $this->payments()->where('user_id', $user->id)->where('status', 'paid')->exists() : false;
        $isLive = $this->status === SessionStatus::Live;
        $isStarted = $this->started_at !== null;

        return [
            'id' => $this->id,
            'group_id' => $this->group_id,
            'instructor_id' => $this->instructor_id,
            'session_number' => $this->session_number,
            'title' => 'Session '.$this->session_number,
            'session_type' => $this->session_type,
            'session_type_label' => $this->session_type === SessionType::Paid ? 'Paid Session' : 'Group Session',
            'status' => $this->status,
            'scheduled_at' => $this->scheduled_at,
            'date' => $this->scheduled_at?->format('D, M d') ?? null,
            'time' => $this->scheduled_at?->format('h:i A') ?? null,
            'started_at' => $this->started_at,
            'ended_at' => $this->ended_at,
            'duration_minutes' => $this->duration_minutes,
            'jitsi_room_name' => $this->jitsi_room_name,
            'jitsi_jwt_issued_at' => $this->jitsi_jwt_issued_at,
            'session_metadata' => $this->session_metadata,
            'price' => data_get($this->session_metadata, 'price', $settings->session_price),
            'formatted_price' => number_format(data_get($this->session_metadata, 'price', $settings->session_price), 0).' EGP',
            'created_at' => $this->created_at,
            'updated_at' => $this->updated_at,
            'seat_number' => $this->payments()->where('status', 'paid')->count(),
            'seat_total' => $settings->group_max_members,
            'seat_remaining' => $settings->group_max_members - $this->payments()->where('status', 'paid')->count(),
            'occupancy_text' => $this->payments()->where('status', 'paid')->count().' Taken / '.$settings->group_max_members,
            'is_booked' => $isBooked,
            'is_locked' => ! ($isLive || $isStarted),
            'started_ago_text' => $isStarted ? 'The session started '.$this->started_at->diffForHumans() : null,
            'instructor_name' => $this->instructor?->display_name,
            'instructor_photo' => $this->instructor?->avatar_url,
            'group' => new GroupResource($this->whenLoaded('group')),
            'instructor' => new UserResource($this->whenLoaded('instructor')),
            'attendances' => SessionAttendanceResource::collection($this->whenLoaded('attendances')),
            'breakout_rooms' => BreakoutRoomResource::collection($this->whenLoaded('breakoutRooms')),
        ];
    }
}

File: Solid.Api/Features/Sessions/SessionResource.cs
using Solid.Api.Common;
using Solid.Api.Database.Entities;

public static class SessionResource
{
    public static object From(TherapySession session, decimal price = 0, long? userId = null)
    {
        var metadata = JsonPayload.ParseObject(session.SessionMetadata);
        var title = Convert.ToString(metadata.GetValueOrDefault("title"));
        var maxParticipants = Convert.ToString(metadata.GetValueOrDefault("max_participants"));
        var userAttendance = userId.HasValue
            ? session.Attendances.FirstOrDefault(attendance => attendance.UserId == userId.Value)
            : null;
        var isBooked = userId.HasValue && session.Payments.Any(payment =>
            payment.UserId == userId.Value &&
            payment.Status == "paid");
        var attendanceStatus = userId.HasValue && (isBooked || userAttendance is not null)
            ? userAttendance?.WasPresent == true ? "attended" : "absent"
            : null;
        var paidParticipants = session.Payments
            .Where(payment => payment.Status == "paid")
            .Select(payment => payment.UserId)
            .Distinct()
            .Count();
        var attendedParticipants = session.Attendances.Count(attendance => attendance.WasPresent);
        var currentParticipants = Math.Max(paidParticipants, attendedParticipants);

        return new
        {
            id = session.Id,
            group_id = session.SubstanceCategoryId,
            group_name = session.SubstanceCategory?.NameAr,
            group_name_ar = session.SubstanceCategory?.NameAr,
            group_name_en = session.SubstanceCategory?.NameEn,
            instructor_id = session.InstructorId,
            instructor_name = session.Instructor?.DisplayName,
            session_number = session.SessionNumber,
            title = string.IsNullOrWhiteSpace(title) ? $"Session {session.SessionNumber}" : title,
            session_type = session.SessionType,
            session_type_label = string.Equals(session.SessionType, "individual", StringComparison.OrdinalIgnoreCase) ? "Individual Session" : "Group Session",
            status = session.Status,
            scheduled_at = EgyptDateTime.Format(session.ScheduledAt),
            date = EgyptDateTime.Date(session.ScheduledAt),
            time = EgyptDateTime.Time(session.ScheduledAt),
            started_at = EgyptDateTime.Format(session.StartedAt),
            ended_at = EgyptDateTime.Format(session.EndedAt),
            duration_minutes = session.DurationMinutes,
            jitsi_room_name = session.JitsiRoomName,
            jitsi_jwt_issued_at = EgyptDateTime.Format(session.JitsiJwtIssuedAt),
            session_metadata = JsonPayload.Parse(session.SessionMetadata),
            max_participants = int.TryParse(maxParticipants, out var parsedMaxParticipants)
                ? MaxParticipants(session, parsedMaxParticipants)
                : MaxParticipants(session, null),
            current_participants = currentParticipants,
            is_full = currentParticipants >= (int.TryParse(maxParticipants, out var fullMaxParticipants)
                ? MaxParticipants(session, fullMaxParticipants)
                : MaxParticipants(session, null)),
            price = price,                              // USE PARAM
            formatted_price = $"{price:0.##} EGP",     // USE PARAM
            created_at = EgyptDateTime.Format(session.CreatedAt),
            updated_at = EgyptDateTime.Format(session.UpdatedAt),
            is_booked = isBooked,
            attendance_status = attendanceStatus,
            is_locked = session.Status != "live"
        };
    }

    private static int MaxParticipants(TherapySession session, int? metadataMaxParticipants)
    {
        if (string.Equals(session.SessionType, "individual", StringComparison.OrdinalIgnoreCase))
        {
            return 1;
        }

        return metadataMaxParticipants is > 0
            ? Math.Min(metadataMaxParticipants.Value, 15)
            : 15;
    }
}
